.circleci/config.yml

```diff
@@ -302,6 +315,122 @@ jobs:
+      - run:
+          name: Maybe merge an automatic PR into develop
+          command: ./tools/circleci/github_scripts/attempt_merge_pr.sh "merge-main-to-develop" "develop" "${GITHUB_TAGGING_TOKEN}"
+
+  build_docker_image:
+    docker:
+      - image: cimg/python:3.8
+    environment:
+      ECR_IMAGE_URL: 350138855857.dkr.ecr.eu-west-2.amazonaws.com/kedro-builder
+      AWS_REGION: eu-west-2
+    steps:
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - checkout
+      - run:
+          name: Setup AWS CLI
+          command: pip install -U awscli
+      - run:
+          name: Login to AWS ECR
+          command: |
+            aws ecr get-login-password --region ${AWS_REGION} | \
+            docker login --username AWS --password-stdin "${ECR_IMAGE_URL}"
+      - run:
+          name: Build docker images
+          command: ./tools/circleci/docker_build_img/build.sh "." "${ECR_IMAGE_URL}"
+          no_output_timeout: 20m
+      - run:
+          name: Logout from AWS ECR
+          command: docker logout "${ECR_IMAGE_URL}"
+          when: always  # logout even if the previous step has failed
+
+  # This is effectively just a combination of the lint, unit_tests and e2e_tests jobs.
+  # It's used to check that the nightly docker image is working ok and before publishing a release.
+  build_kedro:
+    parameters:
+      python_version:
+        type: string
+    executor:
+      name: docker
+      python_version: <<parameters.python_version>>
+    steps:
+      - setup
+      - run:
+          name: Run linters
+          command: make lint
+      - run:
+          name: Run unit tests
+          command: make test
+      - run:
+          name: Run e2e tests
+          command: make e2e-tests
+
+  publish_kedro:
+    executor:
+      name: docker
+      python_version: "3.7"
+    steps:
+      - setup
+      - configure_ssh_keys
+      - remove_ssh_config
+      - run:
+          name: Check Kedro version
+          command: |
+            KEDRO_VERSION=$(./tools/circleci/github_scripts/kedro_version.py ./kedro)
+
+            if ./tools/circleci/check-no-version-pypi.sh "${KEDRO_VERSION}"
+            then
+                echo "export KEDRO_VERSION=\"${KEDRO_VERSION}\"" >> $BASH_ENV
+            else
+                echo "Error: Kedro version ${KEDRO_VERSION} already exists on PyPI"
+                exit 1
+            fi
+      - run:
+          name: Tag and publish release on Github
+          command: ./tools/circleci/github_scripts/release.sh quantumblacklabs kedro ${GITHUB_TAGGING_TOKEN} ${KEDRO_VERSION}
+      - run:
+          name: Publish to PyPI
+          command: |
+            make package
+            python -m pip install twine -U
+            python -m twine upload --repository-url ${TWINE_REPOSITORY_URL} dist/*
+      - run:
+          name: Trigger Read The Docs build
+          command: |
+            ./tools/circleci/rtd-build.sh ${RTD_TOKEN} stable
```

Comment:
Are these env variables set on this repo's CCI or not yet? We should be careful not to leave them behind.

Reply:
I'm going to chat with @idanov to make sure we get all of them here.

Reply:
All the environment variables have now been ported over to this repo.